src/main/java/Duke.java

```diff
@@ -1,10 +1,372 @@
+        return taskDescription.trim().length() > 0;
+    }
+
+    private int getTaskNumFromMarkCommand(String commandInfo) throws DukeException {
+        if (isCommandInfoPresent(commandInfo)) {
+            try {
+                return Integer.parseInt(commandInfo);
+            } catch (NumberFormatException e) {
+                throw new DukeException("Please enter a valid task number to be marked as done!");
+            }
+        } else {
+            throw new DukeException("Please enter a task number to be marked as done!");
+        }
+    }
+
+    private int getTaskNumFromUnmarkCommand(String commandInfo) throws DukeException {
+        if (isCommandInfoPresent(commandInfo)) {
+            try {
+                return Integer.parseInt(commandInfo);
+            } catch (NumberFormatException e) {
+                throw new DukeException("Please enter a valid task number to be marked as not done yet!");
+            }
+        } else {
+            throw new DukeException("Please enter a task number to be marked as not done yet!");
+        }
+    }
+
+    private int getTaskNumFromDeleteCommand(String commandInfo) throws DukeException {
+        if (isCommandInfoPresent(commandInfo)) {
+            try {
+                return Integer.parseInt(commandInfo);
+            } catch (NumberFormatException e) {
+                throw new DukeException("Please enter a valid task number to be deleted!");
+            }
+        } else {
+            throw new DukeException("Please enter a task number to be deleted!");
+        }
+    }
+
+    private String getTaskDescriptionFromToDoCommand(String commandInfo) throws DukeException {
+        if (isCommandInfoPresent(commandInfo)) {
+            return commandInfo;
+        } else {
+            throw new DukeException("The description of a todo cannot be empty!");
+        }
+    }
+
+    private String processTaskInfoFromDeadlineCommand(String commandInfo) throws DukeException {
+        if (!isCommandInfoPresent(commandInfo)) {
+            throw new DukeException("INCOMPLETE COMMAND"
+                    + System.lineSeparator() + "\t"
+                    + "The description and date/time of a deadline cannot be empty!"
+                    + System.lineSeparator() + "\t"
+                    + "[Note: Enter /by before specifying the date/time]");
+        } else {
+            String[] taskInfoParts = commandInfo.split("/", 2);
+
+            if (taskInfoParts.length != 2) {
+                throw new DukeException("INCOMPLETE COMMAND"
+                        + System.lineSeparator() + "\t"
+                        + "The date/time of a deadline cannot be empty!"
+                        + System.lineSeparator() + "\t"
+                        + "[Note: Enter /by before specifying the date/time]");
+            } else {
+                return getTaskInfoFromDeadlineCommand(commandInfo, taskInfoParts);
+            }
+        }
+    }
```

Comment:
Generally, your code is very well indented and its done with good coding standards. However, some of your methods are missing javadoc comments, and it would be good to add more!